scripts/helpers/componentNames.js

```diff
@@ -0,0 +1,41 @@
+module.exports = [
+  {
+    name: "datacollector",
+    exportName: "createDataCollector",
+    filePath: "DataCollector"
+  },
+  {
+    name: "decisioningengine",
+    exportName: "createDecisioningEngine",
+    filePath: "DecisioningEngine"
+  },
+  {
+    name: "activitycollector",
+    exportName: "createActivityCollector",
+    filePath: "ActivityCollector"
+  },
+  { name: "identity", exportName: "createIdentity", filePath: "Identity" },
+  { name: "audiences", exportName: "createAudiences", filePath: "Audiences" },
+  {
+    name: "personalization",
+    exportName: "createPersonalization",
+    filePath: "Personalization"
+  },
+  { name: "context", exportName: "createContext", filePath: "Context" },
+  { name: "privacy", exportName: "createPrivacy", filePath: "Privacy" },
+  {
+    name: "eventmerge",
+    exportName: "createEventMerge",
+    filePath: "EventMerge"
+  },
+  {
+    name: "libraryinfo",
+    exportName: "createLibraryInfo",
+    filePath: "LibraryInfo"
+  },
+  {
+    name: "machinelearning",
+    exportName: "createMachineLearning",
+    filePath: "MachineLearning"
+  }
+];
```

Comment:
Like Joe has mentioned, I don't like having to update this file anytime we create a new component. One way we could do this is import "src/core/componentCreators.js" within scripts. To do this we'd have to make sure we are using Node version >= 13 because currently scripts are written with require and the rest of the code is written with import. Ideally we should move all the code to use import and require a version of Node that supports it. Then we could loop through the component array and grab the name of each one.

Reply:
Deleted the componentNames.js and dynamically creating the list from componentCreators. customBuild.js script reads componentCreators.js, uses a regular expression to find all occurrences of create followed by an uppercase letter and any number of word characters (which should match the names of the component creator functions), formats the component names for export, and writes them to componentNames.js.